src/Microsoft.OData.Core/ODataWriter.cs

```diff
@@ -51,7 +51,10 @@ public ODataWriter Write(ODataResourceSet resourceSet, Action nestedAction)
         /// <summary> Asynchronously start writing a resource set. </summary>
         /// <returns>A task instance that represents the asynchronous write operation.</returns>
         /// <param name="resourceSet">The resource set or collection to write.</param>
-        public abstract Task WriteStartAsync(ODataResourceSet resourceSet);
+        public virtual Task WriteStartAsync(ODataResourceSet resourceSet)
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(() => this.WriteStart(resourceSet));
+        }

```

Comment:
all the changes in this class looks not related to Deep updates? Right? 

However, i am ok to make these changes.

Reply:
Right; this was the separate issue to support async. I added it because I was in the code, but you're right, this is not related to deep updates.

---
In reply to: [348191583](https://github.com/OData/odata.net/pull/1585#discussion_r348191583) [](ancestors = 348191583)